marimo/_runtime/virtual_file.py

```diff
@@ -135,7 +136,14 @@ def add(self, virtual_file: VirtualFile) -> None:
         shm.buf[: len(buffer)] = buffer
         # we can safely close this shm, since we don't need to access its
         # buffer; we do need to keep it around so we can unlink it later
-        shm.close()
+        if sys.platform != "win32":
+            # don't call close() on Windows, due to a bug in the Windows
+            # Python implementation. On Windows, close() actually unlinks
+            # (destroys) the shared_memory:
+            # https://stackoverflow.com/questions/63713241/segmentation-fault-using-python-shared-memory/63717188#63717188
+            shm.close()
```

Comment:
will this be a memory leak on windows?

Reply:
good catch. I checked and didn't find any leaks. but added `close()` calls on removal to be safe